File: PostmanTest/EnclosureTests/HtmlEnclosureTest.vb
Imports System.Net.Mail
Imports Postman

'''<summary>
'''This is a test class for HtmlEnclosure and is intended to contain all HtmlEnclosure and EmbeddedResource Unit Tests
'''</summary>
<TestClass()> _
Public Class HtmlEnclosureTest

    '''<summary>
    '''A test for HtmlEnclosure Constructor
    '''</summary>
    <TestMethod()> _
    Public Sub HtmlEnclosureConstructorTest()
        'Arrange
        Dim expectedAltViewsCount As Integer = 1
        Dim expectedLinkedResourceCount As Integer = 0
        Dim expectedContent As String = "expected<b>Content<b/>"
        Dim expectedEncoding As System.Text.Encoding = System.Text.Encoding.UTF8
        Dim expectedMimeTpye As String = Net.Mime.MediaTypeNames.Text.Html
        Dim msg As New MailMessage()
        Dim target As IEnclosure = New Enclosure.Html(expectedContent)

        'Act
        target.Include(msg)

        'Assert
        Assert.AreEqual(expectedAltViewsCount, msg.AlternateViews.Count)
        Assert.AreEqual(expectedContent, New IO.StreamReader(msg.AlternateViews(0).ContentStream).ReadToEnd)
        Assert.AreEqual(expectedEncoding.WebName, msg.AlternateViews(0).ContentType.CharSet)
        Assert.AreEqual(expectedMimeTpye, msg.AlternateViews(0).ContentType.MediaType)
        Assert.AreEqual(expectedLinkedResourceCount, msg.AlternateViews(0).LinkedResources.Count)
    End Sub

    '''<summary>
    '''A test for HtmlEnclosure Constructor
    '''</summary>
    <TestMethod()> _
    Public Sub HtmlEnclosureConstructorTest1()
        'Arrange
        Dim expectedAltViewsCount As Integer = 1
        Dim expectedLinkedResourceCount As Integer = 1
        Dim expectedContent As String = "expected<b>Content<b/>"
        Dim expectedEncoding As System.Text.Encoding = System.Text.Encoding.UTF8
        Dim expectedMimeTpye As String = Net.Mime.MediaTypeNames.Text.Html
        Dim expectedContentId As String = "expectedContentId"
        Dim expectedContentType As New Net.Mime.ContentType("img/jpeg")
        Dim expectedEmbeddedResource As IEmbeddedResource = New EmbeddedResource.EmbeddedResource(New System.IO.MemoryStream(), expectedContentType, expectedContentId)
        Dim resList As New List(Of IEmbeddedResource)
        resList.Add(expectedEmbeddedResource)
        Dim msg As New MailMessage()
        Dim target As IEnclosure = New Enclosure.Html(expectedContent, resList)

        'Act
        target.Include(msg)

        'Assert
        Assert.AreEqual(expectedAltViewsCount, msg.AlternateViews.Count)
        Assert.AreEqual(expectedContent, New IO.StreamReader(msg.AlternateViews(0).ContentStream).ReadToEnd)
        Assert.AreEqual(expectedEncoding.WebName, msg.AlternateViews(0).ContentType.CharSet)
        Assert.AreEqual(expectedMimeTpye, msg.AlternateViews(0).ContentType.MediaType)

        Assert.AreEqual(expectedLinkedResourceCount, msg.AlternateViews(0).LinkedResources.Count)
        Assert.AreEqual(expectedContentId, msg.AlternateViews(0).LinkedResources(0).ContentId)
        Assert.AreEqual(expectedContentType, msg.AlternateViews(0).LinkedResources(0).ContentType)
    End Sub

End Class

File: Postman.Tests/Enclosure/HtmlEnclosureTest.cs
namespace Postman.Tests
{
    using System.Collections.Generic;
    using System.IO;
    using System.Net.Mail;
    using System.Net.Mime;
    using Xunit;

    /// <summary>
    /// This is a test class for HtmlEnclosure and is intended to contain all HtmlEnclosure and EmbeddedResource Unit Tests
    /// </summary>
    public class HtmlEnclosureTest
    {
        /// <summary>
        /// A test for HtmlEnclosure Constructor
        /// </summary>
        [Fact]
        public void HtmlEnclosureConstructorTest()
        {
            // Arrange
            int expectedAltViewsCount = 1;
            int expectedLinkedResourceCount = 0;
            string expectedContent = "expected<b>Content<b/>";
            System.Text.Encoding expectedEncoding = System.Text.Encoding.UTF8;
            string expectedMimeTpye = MediaTypeNames.Text.Html;
            MailMessage msg = new MailMessage();
            IEnclosure target = new Enclosure.Html(expectedContent);

            // Act
            target.Include(msg);

            // Assert
            Assert.Equal(expectedAltViewsCount, msg.AlternateViews.Count);
            Assert.Equal(expectedContent, new StreamReader(msg.AlternateViews[0].ContentStream).ReadToEnd());
            Assert.Equal(expectedEncoding.WebName, msg.AlternateViews[0].ContentType.CharSet);
            Assert.Equal(expectedMimeTpye, msg.AlternateViews[0].ContentType.MediaType);
            Assert.Equal(expectedLinkedResourceCount, msg.AlternateViews[0].LinkedResources.Count);
        }

        /// <summary>
        /// A test for HtmlEnclosure Constructor
        /// </summary>
        [Fact]
        public void HtmlEnclosureWithEmbeddedResource()
        {
            // Arrange
            int expectedAltViewsCount = 1;
            int expectedLinkedResourceCount = 1;
            string expectedContent = "expected<b>Content<b/>";
            System.Text.Encoding expectedEncoding = System.Text.Encoding.UTF8;
            string expectedMimeTpye = MediaTypeNames.Text.Html;
            string expectedContentId = "expectedContentId";
            ContentType expectedContentType = new ContentType("img/jpeg");
            IEmbeddedResource expectedEmbeddedResource = new EmbeddedResource.EmbeddedResource(new System.IO.MemoryStream(), expectedContentType, expectedContentId);
            ICollection<IEmbeddedResource> resList = new List<IEmbeddedResource>();
            resList.Add(expectedEmbeddedResource);
            MailMessage msg = new MailMessage();
            IEnclosure target = new Enclosure.Html(expectedContent, resList);

            // Act
            target.Include(msg);

            // Assert
            Assert.Equal(expectedAltViewsCount, msg.AlternateViews.Count);
            Assert.Equal(expectedContent, new System.IO.StreamReader(msg.AlternateViews[0].ContentStream).ReadToEnd());
            Assert.Equal(expectedEncoding.WebName, msg.AlternateViews[0].ContentType.CharSet);
            Assert.Equal(expectedMimeTpye, msg.AlternateViews[0].ContentType.MediaType);

            Assert.Equal(expectedLinkedResourceCount, msg.AlternateViews[0].LinkedResources.Count);
            Assert.Equal(expectedContentId, msg.AlternateViews[0].LinkedResources[0].ContentId);
            Assert.Equal(expectedContentType, msg.AlternateViews[0].LinkedResources[0].ContentType);
        }
    }
}